### PR TITLE
Set DefaultHandler in RouteBuilder

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Issue #294 

@pranavkm

### CODE REVIEW

**Comment by davidfowl** [comment]
This so needs a test though

**Comment by ajaybhargavb** [comment]
Added.

File: test/Microsoft.AspNetCore.Routing.Tests/RouteBuilderTest.cs [hunk 1]
```diff
@@ -11,6 +12,27 @@ namespace Microsoft.AspNetCore.Routing
...
+        public void Ctor_SetsPropertyValues()
+        {
+            // Arrange
+            var applicationServicesMock = new Mock<IServiceProvider>();
+            applicationServicesMock.Setup(s => s.GetService(typeof(RoutingMarkerService))).Returns(Mock.Of<RoutingMarkerService>);
+            var applicationBuilderMock = new Mock<IApplicationBuilder>();
+            var applicationServices = applicationServicesMock.Object;
+            applicationBuilderMock.Setup(a => a.ApplicationServices).Returns(applicationServices);
+            var applicationBuilder = applicationBuilderMock.Object;
+            var defaultHandler = Mock.Of<IRouter>();
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Routing.Tests/RouteBuilderTest.cs]: Can we just use `IServiceCollection` here instead?

**Comment by rynowak** [comment]
:shipit:

File: test/Microsoft.AspNetCore.Routing.Tests/RouteBuilderTest.cs (hunk 1 above)
Comment (ajaybhargavb) [inline test/Microsoft.AspNetCore.Routing.Tests/RouteBuilderTest.cs]: updated the test

**Pull request merged**